app/src/main/java/com/quran/labs/androidquran/ui/util/ToastCompat.java

```diff
@@ -0,0 +1,83 @@
+package com.quran.labs.androidquran.ui.util;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.graphics.Color;
+import android.graphics.drawable.Drawable;
+import android.os.Build;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.StringRes;
+import androidx.appcompat.content.res.AppCompatResources;
+import androidx.core.content.ContextCompat;
+import androidx.core.graphics.drawable.DrawableCompat;
+
+import com.quran.labs.androidquran.R;
+import com.quran.labs.androidquran.util.QuranSettings;
+
+public class ToastCompat {
+
+  private static final int ANDROID_R = 30;
+
+  public static Toast makeText(@NonNull Context context, @StringRes int messageRes, int duration) {
+    return makeText(context, context.getString(messageRes), duration);
+  }
+
+  public static Toast makeText(@NonNull Context context, CharSequence message, int duration) {
+    @SuppressLint("ShowToast") Toast toast = Toast.makeText(context, message, duration);
+    if (Build.VERSION.SDK_INT < ANDROID_R) {
+      // It's unclear what behavior `Toast.setView()` has on R and above, hence the guard.
+      toast.setView(createToastView(context, message));
+    }
+    return toast;
+  }
+
+  private static View createToastView(Context context, CharSequence message) {
+    View toastView = LayoutInflater.from(context).inflate(R.layout.toast, null, false);
+
+    QuranSettings settings = QuranSettings.getInstance(context);
+    Drawable tintedBackground = createTintedBackground(context, settings.isNightMode());
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+      toastView.setBackground(tintedBackground);
+    } else {
+      toastView.setBackgroundDrawable(tintedBackground);
+    }
+
+    TextView textView = toastView.findViewById(android.R.id.message);
```

Comment:
I continued to use the same IDs that the default Toast layout used.

Reply:
let's play it safe here and do nothing if it's null just in case Samsung devices do something odd

Reply:
This will never be null as we inflate a layout `R.layout.toast` which uses this ID. I can rename the ID?

I wasn't sure if the system would rely on this ID being something particular, but it shouldn't.

Reply:
ah right you're inflating your own layout - never mind then, this is fine